packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/EnrichmentService.java

```diff
@@ -0,0 +1,228 @@
+import software.uncharted.terarium.hmiserver.models.dataservice.document.DocumentAsset;
+import software.uncharted.terarium.hmiserver.models.dataservice.model.Model;
+import software.uncharted.terarium.hmiserver.models.dataservice.modelparts.ModelMetadata;
+import software.uncharted.terarium.hmiserver.models.dataservice.modelparts.ModelParameter;
+import software.uncharted.terarium.hmiserver.models.dataservice.modelparts.semantics.Observable;
+import software.uncharted.terarium.hmiserver.models.dataservice.modelparts.semantics.State;
+import software.uncharted.terarium.hmiserver.models.dataservice.modelparts.semantics.Transition;
+import software.uncharted.terarium.hmiserver.models.dataservice.regnet.RegNetVertex;
+import software.uncharted.terarium.hmiserver.models.task.CompoundTask;
+import software.uncharted.terarium.hmiserver.models.task.TaskRequest;
+import software.uncharted.terarium.hmiserver.models.task.TaskResponse;
+import software.uncharted.terarium.hmiserver.models.task.TaskStatus;
+import software.uncharted.terarium.hmiserver.service.data.DKGService;
+import software.uncharted.terarium.hmiserver.service.data.DocumentAssetService;
+import software.uncharted.terarium.hmiserver.service.data.ModelService;
+import software.uncharted.terarium.hmiserver.service.notification.NotificationGroupInstance;
+import software.uncharted.terarium.hmiserver.service.notification.NotificationService;
+import software.uncharted.terarium.hmiserver.service.tasks.TaskService;
+import software.uncharted.terarium.hmiserver.service.tasks.TaskUtilities;
+import software.uncharted.terarium.hmiserver.utils.rebac.Schema;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class EnrichmentService {
+
+	private final ClientEventService clientEventService;
+	private final CurrentUserService currentUserService;
+	private final DKGService dkgService;
+	private final DocumentAssetService documentService;
+	private final ModelService modelService;
+	private final NotificationService notificationService;
+	private final TaskService taskService;
+
+	@Value("${terarium.extractionService.poolSize:10}")
+	private int POOL_SIZE;
+
+	@Data
+	private static class Properties {
+
+		private final UUID modelId;
+	}
+
+	/**
+	 * Enriches the model with the data from the document.
+	 *
+	 * @param projectId the project ID
+	 * @param documentId the document ID
+	 * @param modelId the model ID
+	 * @param permission the permission
+	 * @return the enriched model
+	 */
+	public Future<Model> modelWithDocument(UUID projectId, UUID documentId, UUID modelId, Schema.Permission permission) {
+		// Create the notification group
+		final NotificationGroupInstance<Properties> notificationInterface = new NotificationGroupInstance<>(
+			clientEventService,
+			notificationService,
+			ClientEventType.ENRICHMENT_MODEL,
+			projectId,
+			new Properties(modelId)
+		);
+
+		ExecutorService executor = Executors.newFixedThreadPool(POOL_SIZE);
```

Comment:
Put the pool on the class. You want a single persistent pool.

Reply:
Isn't what's above on L49 ?
https://github.com/DARPA-ASKEM/terarium/pull/5583/files/30f21d6d69df3f8dac13ea8dcb739fb285d08a0a#diff-c904c511e871ff90a3ae8a64d0596ab5cc8bcd9c9125fc9e36d6494ca0aff2c3R49-R50